lib/msal-core/dist/msal.js

```diff
@@ -2785,14 +2804,34 @@ var UserAgentApplication = /** @class */ (function () {
         if (this._user) {
             return this._user;
         }
-        // frame is used to get idToken
-        var rawIdToken = this._cacheStorage.getItem(Constants_1.Constants.idTokenKey);
-        var rawClientInfo = this._cacheStorage.getItem(Constants_1.Constants.msalClientInfo);
-        if (!Utils_1.Utils.isEmpty(rawIdToken) && !Utils_1.Utils.isEmpty(rawClientInfo)) {
-            var idToken = new IdToken_1.IdToken(rawIdToken);
-            var clientInfo = new ClientInfo_1.ClientInfo(rawClientInfo);
-            this._user = User_1.User.createUser(idToken, clientInfo, this.authority);
-            return this._user;
+        var adalIdToken = this._cacheStorage.getItem(Constants_1.Constants.adalIdToken);
+        if (!Utils_1.Utils.isEmpty(adalIdToken)) {
```

Comment:
If MSAL's idtoken exists then this._user object will be present and it will return before only. At this line https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/370306f1a044784bd4f5d311da07a0140cb45373/lib/msal-core/src/UserAgentApplication.ts#L1372

if (this._user) {
      return this._user;
    }